service/src/main/java/bio/terra/pipelines/service/QuotasService.java

```diff
@@ -75,4 +75,32 @@ public UserQuota updateQuotaConsumed(UserQuota userQuota, int newQuotaConsumed)
     userQuota.setQuotaConsumed(newQuotaConsumed);
     return userQuotasRepository.save(userQuota);
   }
+
+  /**
+   * This method updates the quota limit for a given user quota. This should only be called from the
+   * Admin Controller
+   *
+   * @param userQuota - the user quota to update
+   * @param newQuotaLimit - the new quota limit
+   * @return - the updated user quota
+   */
+  public UserQuota updateQuotaLimit(UserQuota userQuota, int newQuotaLimit) {
+    if (newQuotaLimit < userQuota.getQuotaConsumed()) {
+      throw new InternalServerErrorException(
+          String.format(
+              "New quota limit: %d, is less than the quota consumed: %d",
+              newQuotaLimit, userQuota.getQuotaConsumed()));
+    }
+    userQuota.setQuota(newQuotaLimit);
+    return userQuotasRepository.save(userQuota);
+  }
+
+  /**
+   * @param userId - the user id
+   * @param pipelineName - the pipeline name
+   * @return - true if the user quota is present, false otherwise
+   */
+  public boolean isUserQuotaPresent(String userId, PipelinesEnum pipelineName) {
```

Comment:
since this doesn't do any other authz checking, can we mark this as also must be called by an admin controller? (just in documentation / function name, as above)

Reply:
that's a good point, sounds good 👍